From d856718e69a32aedd37e8dc722735e6c85c85224 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 24 Apr 2020 11:30:50 +0800
Subject: [PATCH] ARM: dts: rv1126: Move rgb dt node as a sub node of grf

Move rgb dt node under grf, so it can set LCD_IO_BYPASS
bit in grf.

Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
Change-Id: I50b463f37010d1273f4a8e6d299d419877cb3f9c
---
 arch/arm/boot/dts/rv1126.dtsi | 48 +++++++++++++++++------------------
 1 file changed, 23 insertions(+), 25 deletions(-)

diff --git a/arch/arm/boot/dts/rv1126.dtsi b/arch/arm/boot/dts/rv1126.dtsi
index 7e3f5e138721..637ad95d4108 100644
--- a/arch/arm/boot/dts/rv1126.dtsi
+++ b/arch/arm/boot/dts/rv1126.dtsi
@@ -209,30 +209,6 @@
 		};
 	};
 
-	rgb: rgb {
-		compatible = "rockchip,rv1126-rgb";
-		pinctrl-names = "default";
-		pinctrl-0 = <&lcdc_ctl>;
-		status = "disabled";
-
-		ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				rgb_in_vop: endpoint@0 {
-					reg = <0>;
-					remote-endpoint = <&vop_out_rgb>;
-				};
-			};
-
-		};
-	};
-
 	thermal_zones: thermal-zones {
 		cpu_thermal: cpu-thermal {
 			polling-delay-passive = <20>; /* milliseconds */
@@ -282,8 +258,30 @@
 	};
 
 	grf: syscon@fe000000 {
-		compatible = "rockchip,rv1126-grf", "syscon";
+		compatible = "rockchip,rv1126-grf", "syscon", "simple-mfd";
 		reg = <0xfe000000 0x20000>;
+
+		rgb: rgb {
+			compatible = "rockchip,rv1126-rgb";
+			status = "disabled";
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@0 {
+					reg = <0>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					rgb_in_vop: endpoint@0 {
+						reg = <0>;
+						remote-endpoint = <&vop_out_rgb>;
+					};
+				};
+
+			};
+		};
 	};
 
 	pmugrf: syscon@fe020000 {
-- 
2.35.3

